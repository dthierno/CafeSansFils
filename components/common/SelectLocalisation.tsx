import {
  Text,
  StyleSheet,
  TouchableOpacity,
  GestureResponderEvent,
} from "react-native";
import { StyleProp, ViewStyle } from "react-native";

import { ChevronDown, MapPin } from "lucide-react-native";

import TYPOGRAPHY from "@/constants/Typography";
import COLORS from "@/constants/Colors";
import SPACING from "@/constants/Spacing";

type SelectLocalisationProps = {
  currentLocalisation: string;
  style?: StyleProp<ViewStyle>;
};

export default function SelectLocalisation({
  currentLocalisation,
  style,
}: SelectLocalisationProps) {
  function handlePress(event: GestureResponderEvent): void {
    console.log("`handlePress` function not implemented.");
  }
  return (
    <TouchableOpacity
      onPress={handlePress}
      style={[style, styles.selectLocalisationContainer]}
      testID="select-localisation-container"
    >
      <MapPin
        width={SPACING.md}
        height={SPACING.md}
        strokeWidth={2.5}
        color={COLORS.subtuleDark}
        testID="map-pin"
      />
      <Text
        style={[TYPOGRAPHY.body.normal.semiBold, styles.localisationText]}
        testID="localisation-text"
      >
        {currentLocalisation}
      </Text>
      <ChevronDown
        width={SPACING.lg}
        height={SPACING.lg}
        strokeWidth={2.5}
        color={COLORS.subtuleDark}
        testID="chevron-down"
      />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  selectLocalisationContainer: {
    flexDirection: "row",
    alignItems: "center",
    gap: SPACING.xs,
    padding: SPACING.xs,
  },
  localisationText: {
    marginLeft: SPACING.xxs,
    color: COLORS.subtuleDark,
  },
});
