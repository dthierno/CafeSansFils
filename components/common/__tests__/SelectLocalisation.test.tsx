import { render, fireEvent, screen } from "@testing-library/react-native";
import SelectLocalisation from "@/components/common/SelectLocalisation";
import COLORS from "@/constants/Colors";

describe("SelectLocalisation", () => {
  const mockProps = {
    currentLocalisation: "Montreal, QC",
  };

  test("renders correctly with the current localisation text", () => {
    render(<SelectLocalisation {...mockProps} />);

    expect(screen.getByText(mockProps.currentLocalisation)).toBeTruthy();

    const mapPinIcon = screen.getAllByTestId("map-pin");
    const chevronDownIcon = screen.getAllByTestId("chevron-down");

    expect(mapPinIcon).toBeTruthy();
    expect(chevronDownIcon).toBeTruthy();
  });

  test("calls the `handlePress` function when pressed", () => {
    render(<SelectLocalisation {...mockProps} />);
    const button = screen.getByTestId("select-localisation-container");

    // Mock the console log for testing
    const consoleSpy = jest.spyOn(console, "log").mockImplementation(() => {});

    fireEvent.press(button);
    expect(consoleSpy).toHaveBeenCalledWith(
      "`handlePress` function not implemented."
    );

    // Restore the original console log implementation
    consoleSpy.mockRestore();
  });

  test("matches the snapshot", () => {
    const tree = render(<SelectLocalisation {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});
