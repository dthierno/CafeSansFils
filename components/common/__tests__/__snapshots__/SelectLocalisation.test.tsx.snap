// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectLocalisation matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "opacity": 1,
      "padding": 8,
    }
  }
  testID="select-localisation-container"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={16}
    bbWidth={16}
    fill="none"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="#424242"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2.5}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    testID="map-pin"
    vbHeight={24}
    vbWidth={24}
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4282532418,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2.5}
    >
      <RNSVGPath
        d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.5}
        testID="map-pin"
      />
      <RNSVGCircle
        cx="12"
        cy="10"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        r="3"
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.5}
        testID="map-pin"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-SemiBold",
          "fontSize": 12,
        },
        {
          "color": "#424242",
          "marginLeft": 4,
        },
      ]
    }
    testID="localisation-text"
  >
    Montreal, QC
  </Text>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="#424242"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2.5}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    testID="chevron-down"
    vbHeight={24}
    vbWidth={24}
    width={20}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4282532418,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2.5}
    >
      <RNSVGPath
        d="m6 9 6 6 6-6"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.5}
        testID="chevron-down"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
